### PR TITLE
nshlib/[cd|ls|pwd]: add support for local CWD(Current working directory)

### PR DESCRIPTION
**Pull request by anchao**
## Summary

nshlib/[cd|ls|pwd]: add support for local CWD(Current working directory)

This PR will still allow basic shell operations such as cd/ls/pwd to be used even when the environment is disabled.

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

Depends on: https://github.com/apache/nuttx/pull/15120

## Testing

sim/nsh  and enable  CONFIG_DISABLE_ENVIRON

before:
```
$ ./build/nuttx 

NuttShell (NSH) NuttX-10.4.0
nsh> ls
nsh: ls: missing required argument(s)
nsh> cd
nsh: cd: command not found
nsh> pwd
nsh: pwd: command not found
nsh> ls /
/:
 bin/
 data/
 dev/
 etc/
 proc/
 tmp/
nsh> 
```
after:

```
$ ./build/nuttx 

NuttShell (NSH) NuttX-10.4.0
nsh> ls
/:
 bin/
 data/
 dev/
 etc/
 proc/
 tmp/
nsh> cd
nsh> cd /data
nsh> pwd
/data
nsh> ls /
/:
 bin/
 data/
 dev/
 etc/
 proc/
 tmp/
nsh> 

```

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



This PR does **not** fully meet the NuttX requirements.  While it provides a good starting point, it lacks crucial information in several sections.

Here's a breakdown of what's missing and how to improve it:

* **Summary:**
    * **Missing:** A clear explanation of *why* this change is necessary.  Simply stating it allows basic shell operations when the environment is disabled isn't enough.  What problem does this solve? What use case does it enable?  Why is it important to have these commands work without the environment?
    * **Missing:**  What functional part of the code is changed? Be specific. Mention the files modified (e.g., `nshlib/builtin_cd.c`, etc.).
    * **Missing:** How *exactly* does the change work?  Does it introduce a local CWD variable within the shell? How is it managed? What code paths are affected?

* **Impact:**
    * **Incomplete:**  Saying it "depends on" another PR isn't sufficient.  You need to explain the impact *of this PR itself*, assuming the other PR is merged.  Address *all* the points:
        * **Is new feature added?**  If these commands didn't work at all before, then yes.
        * **Impact on user:** YES.  Explain that users will now be able to use `cd`, `ls`, and `pwd` even with `CONFIG_DISABLE_ENVIRON` enabled.
        * **Impact on build:** Likely NO, but confirm.
        * **Impact on hardware:**  Likely NO, but confirm.
        * **Impact on documentation:**  YES. If the behavior changes, the documentation needs updating to reflect this new functionality.  Mention whether you have included documentation updates in this PR.
        * **Impact on security:**  Consider potential security implications. Does this change expand the attack surface in any way?  Likely NO, but analyze and justify.
        * **Impact on compatibility:**  Potentially YES. Does this change break any existing scripts or user expectations? Likely NO, but consider it.

* **Testing:**
    * **Incomplete:**  Provide more details about your build host: OS version, compiler version, etc.
    * **Incomplete:** Be specific about your target.  "sim/nsh" isn't enough. Which simulator? What configuration? (e.g., `qemu-system-arm -M lm3s6965evb -kernel build/nuttx`).
    * **Insufficient testing:** While the logs show basic functionality, they are not comprehensive. Test edge cases and error conditions:
        * `cd` into non-existent directories.
        * `cd` into files.
        * `cd` with relative paths (`cd ../..`).
        * `ls` on empty directories.
        * `ls` with invalid paths.


By addressing these points, you can significantly improve the quality of your PR and make it easier for reviewers to understand and approve your changes.  A well-written PR increases the likelihood of it being merged quickly and smoothly.

File: nshlib/nsh_console.c [hunk 1]
```diff
@@ -442,6 +442,13 @@ FAR struct console_stdio_s *nsh_newconsole(bool isctty)
       /* Initialize the input stream */
 
       INFD(pstate)               = STDIN_FILENO;
+
+      /* Initialize current working directory */
+
+#ifdef CONFIG_DISABLE_ENVIRON
+      strncpy(pstate->cn_vtbl.cwd, CONFIG_LIBC_HOMEDIR,
```
Comment (xiaoxiang781216) [inline nshlib/nsh_console.c:449]: strlcpy

File: nshlib/nsh_envcmds.c [hunk 2]
```diff
@@ -273,6 +273,13 @@ int cmd_cd(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
       nsh_error(vtbl, g_fmtcmdfailed, argv[0], "chdir", NSH_ERRNO);
       ret = ERROR;
     }
+#ifdef CONFIG_DISABLE_ENVIRON
+  else
+    {
+      strncpy(vtbl->cwd, path,
```
Comment (xiaoxiang781216) [inline nshlib/nsh_envcmds.c:279]:
```suggestion
      strlcpy(vtbl->cwd, path,
```

File: nshlib/nsh_envcmds.c [hunk 3]
```diff
@@ -273,6 +273,13 @@ int cmd_cd(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
...
+              strnlen(path, sizeof(vtbl->cwd) - 1) + 1);
```
Comment (xiaoxiang781216) [inline nshlib/nsh_envcmds.c:280]: whynot simply sizeof(vtbl->cwd)

Comment (anchao) [reply nshlib/nsh_envcmds.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]
